src/RandArray.chpl

```diff
@@ -24,7 +24,11 @@ module RandArray {
       [ai in a] if (ai < 0) { ai = -ai; }
       if (aMax > aMin) {
         const modulus = aMax - aMin;
-        [x in a] x = ((x % modulus) + aMin):t;
+        if modulus > 0 {
```

Comment:
this bends my brain a little, maybe we should comment this with the reason why we do this.

Reply:
@pierce314159 and I initially thought this would work, but thinking about it more, we can expand this.

The `if else` handles detection of an integer overflow in `modulus`, `aMax - aMin > 2**64`. In the case that there is an integer overflow, this solution only provides `2**63` of range. This essentially overrides `aMax` to be `aMin + 2**63`

Reply:
We think we got a better solution by converting `uint` and back. This has the benefit of getting the full 64 bits of range and is simpler logic